### PR TITLE
fixed set.__ror__ operator to behave according to type

### PR DESCRIPTION
**Pull request by joohongpark**
I will also add sub, xor and and operations later.

### CODE REVIEW

File: vm/src/builtins/set.rs [hunk 1]
```diff
@@ -485,30 +485,60 @@ impl PySet {
...
-        self.union(other.iterable, vm)
+    fn or(&self, other: PyObjectRef, vm: &VirtualMachine) -> PyResult<PyArithmeticValue<Self>> {
+        if other.isinstance(&vm.ctx.types.set_type)
+            || other.isinstance(&vm.ctx.types.frozenset_type)
+        {
+            let val = PosArgs::new(vec![ArgIterable::try_from_object(vm, other)?]);
+            Ok(PyArithmeticValue::Implemented(self.union(val, vm)?))
+        } else {
+            Ok(PyArithmeticValue::NotImplemented)
+        }
```
Comment (DimitrisJim) [inline vm/src/builtins/set.rs:496]: `SetIterable` does most of this, it just doesn't return the right things. I think taking inspiration from what `PyInt` does (with `int_op`) would be the best thing here (a function that tries to create a `SetIterable` from the `PyObject` ref and then forwards the call to the appropriate set/frozenset method).

Comment (joohongpark) [reply vm/src/builtins/set.rs:496]: thank you! I will refer to the information you provided and try to correct it! 🙏🏼

Comment (joohongpark) [reply vm/src/builtins/set.rs:496]: I have a question. As you said, I referred to how int_op works in PyInt, which means that PySet should add functions such as set_op to PySet to implement similarly? Or does it mean adding another function to the SetItable structure?

File: vm/src/builtins/dict.rs [hunk 2]
```diff
@@ -946,14 +946,23 @@ trait ViewSetOps: DictView {
         Ok(PySet { inner })
     }
 
-    #[pymethod(name = "__rsub__")]
+    #[pymethod(name = "__sub__")]
```
Comment (youknowone) [inline vm/src/builtins/dict.rs:949]: `__sub__` is handled by below `#[pymethod(magic)]`. Remove this line if you are going to add separated rsub

File: vm/src/builtins/dict.rs [hunk 3]
```diff
@@ -946,14 +946,23 @@ trait ViewSetOps: DictView {
...
     #[pymethod(magic)]
     fn sub(zelf: PyRef<Self>, other: ArgIterable, vm: &VirtualMachine) -> PyResult<PySet> {
         let zelf = Self::to_set(zelf, vm)?;
         let inner = zelf.difference(other, vm)?;
         Ok(PySet { inner })
     }
 
+    #[pymethod(name = "__rsub__")]
```
Comment (youknowone) [inline vm/src/builtins/dict.rs:957]: same here

**Review by youknowone — CHANGES_REQUESTED** [review]

**Review by youknowone — APPROVED** [review]

**Comment by youknowone** [comment]
looks good. please use rebase rather than merge next time.

**Pull request merged**